lib/reservation_manager.rb

```diff
@@ -0,0 +1,204 @@
+      room = find_available_room(range)
+
+      input = parse_reservation_data(checkin_date, checkout_date, room.room_number, room_block_discount)
+      reservation = Hotel::Reservation.new(input)
+
+      room.add_reservation(reservation)
+      add_reservation_to_list(reservation)
+      return reservation
+    end
+
+    # helper method for making room_block / parsing the room_block data
+    def parse_room_block_data(checkin_date, checkout_date, num_rooms, room_numbers)
+
+      return { checkin_date: checkin_date, checkout_date: checkout_date,
+        confirmation_id: "BLOCK#{generate_random_block_id}", num_rooms: num_rooms,
+        room_number: room_numbers, reservations: reservations, total_cost: num_rooms * (calculate_total_cost(checkin_date, checkout_date, @room_cost, @room_block_discount))
+      }
+    end
+
+    #helper method for storing a room_block/ adding it to the list
+    def add_room_block_to_list(room_block)
+      raise ArgumentError.new('Not a valid reservation.') if room_block.class != Hotel::RoomBlock
+      @room_blocks << room_block
+    end
+
+    #helper method for make reservation, which makes room block and its reservations
+    def make_room_block(checkin_date, checkout_date, num_rooms = 2)
+      reservations = []
+      room_numbers = []
+
+      num_rooms.times do
+        reservation = make_reservation(checkin_date, checkout_date, 1, @room_block_discount)
+        reservations << reservation
+        room_numbers << reservation.room_number
+      end
+
+      block_data = parse_room_block_data(checkin_date, checkout_date, num_rooms, room_numbers)
+
+      room_block =  Hotel::RoomBlock.new(block_data)
+      add_room_block_to_list(room_block)
+
+      return room_block
+    end
+
+    #returns total cost for a given reservation
+    def get_total_reservation_cost(given_confirmation_id)
+
+      return @reservations.reduce(0) do |cost, reservation|
+        if reservation.confirmation_id == given_confirmation_id.to_s
+          cost + reservation.total_cost
+        end
+      end
+    end
+
+    #returns array of list of room numbers in all rooms
+    def list_all_rooms_in_hotel
+
+      return @rooms.map { |room| room.room_number }
+    end
+
+    #helper method for listing all available rooms
+    def find_all_available_rooms(given_date_range)
+
+      available_rooms = []
+
+      @rooms.find_all do |room|
+        if (room.is_available?(given_date_range))
+          available_rooms << room.room_number
+        end
+      end
+
+      return "Hotel fully booked for this date range. Try a different date." if available_rooms.empty?
+      return available_rooms
+    end
+
+    #returns array of rooms available today
+    def list_rooms_available_today
+
+      return find_all_available_rooms([Date.today()])
+    end
+
+    #returns array of rooms available for a given date
+    def list_available_rooms(given_date_range)
+
+      return find_all_available_rooms(given_date_range)
+    end
```

Comment:
This method is probably not needed as it is exactly equivalent to calling `find_all_available_rooms` instead.

If we needed to have the `find_all_available_rooms` method also accessible with the name `list_available_rooms` we could write that using this shorthand:

```ruby
alias_method :list_available_rooms, :find_all_available_rooms
```